hail_search/queries/base.py

```diff
@@ -296,7 +300,19 @@ def _parse_sample_data(self, sample_data):
     def _load_filtered_project_hts(self, project_samples, skip_all_missing=False, n_partitions=MAX_PARTITIONS, **kwargs):
         if len(project_samples) == 1:
             project_guid = list(project_samples.keys())[0]
-            project_ht = self._read_table(f'projects/{project_guid}.ht', use_ssd_dir=True)
+            # for variant lookup, project_samples looks like
+            #   {<project_guid>: {<family_guid>: True, <family_guid_2>: True}, <project_guid_2>: ...}
+            # for variant search, project_samples looks like
+            #   {<project_guid>: {<family_guid>: [<sample_data>, <sample_data>, ...], <family_guid_2>: ...}, <project_guid_2>: ...}
```

Comment:
I think it would be neater to change `project_samples` to include some sort of sample type - maybe change the key from project_guid to a project/sample_type tuple. The you could just do the `hfs.exists` call in the variant lookup explicitly instead of running it for everything even when we deterministically know the sample type

Reply:
I agree about adding `sample_type` to the `project_samples` dict. I'm going to leave the variant lookup `project_samples` structure as is because it will change when the lookup table also has sample_type!

Reply:
So I am not comfortable with the change to the `_read_table` behavior in this PR as it is now, I think it adds an inefficiency on every search to check whether or not the table exists, which is unnecessary since for all non-lookup searches we explicitly set the sample type. I think the cleanest way to solve this is to  change the `project_samples` to have the sample_type set for lookup as well, but if you want to leave it as is for now then I think you should move the `hfs.exists` call into the `if isinstance(first_family_samples, bool)` block to explicitly set the sample type, with a comment indicating that this is a temporary change until the lookup table can be refactored

Reply:
You're right about the extra inefficiency of `hfs.exists`. I took your advice and changed `project_samples` to include `sample_type` in the lookup flow